airflow/operators/python.py

```diff
@@ -145,6 +149,145 @@ def execute_callable(self):
+          ...
+          task_id__20
+        """
+        dag = dag or DagContext.get_current_dag()
+        if not dag or task_id not in dag.task_ids:
+            return task_id
+        core = re.split(r'__\d+$', task_id)[0]
+        suffixes = sorted(
+            [int(re.split(r'^.+__', task_id)[1])
+             for task_id in dag.task_ids
+             if re.match(rf'^{core}__\d+$', task_id)]
+        )
+        if not suffixes:
+            return f'{core}__1'
+        return f'{core}__{suffixes[-1] + 1}'
+
+    @staticmethod
+    def validate_python_callable(python_callable):
+        """
+        Validate that python callable can be wrapped by operator.
+        Raises exception if invalid.
+
+        :param python_callable: Python object to be validated
+        :raises: TypeError, AirflowException
+        """
+        if not callable(python_callable):
+            raise TypeError('`python_callable` param must be callable')
+        if 'self' in signature(python_callable).parameters.keys():
+            raise AirflowException('@task does not support methods')
+
+    def execute(self, context: Dict):
+        return_value = self.python_callable(*self.op_args, **self.op_kwargs)
+        self.log.debug("Done. Returned value was: %s", return_value)
+        if not self.multiple_outputs:
+            return return_value
+        if isinstance(return_value, dict):
+            for key in return_value.keys():
+                if not isinstance(key, str):
+                    raise AirflowException('Returned dictionary keys must be strings when using '
+                                           f'multiple_outputs, found {key} ({type(key)}) instead')
+            for key, value in return_value.items():
+                self.xcom_push(context, key, value)
+        else:
+            raise AirflowException(f'Returned output was type {type(return_value)} expected dictionary '
+                                   'for multiple_outputs')
+        return return_value
+
+
+def task(python_callable: Optional[Callable] = None, multiple_outputs: bool = False, **kwargs):
+    """
+    Python operator decorator. Wraps a function into an Airflow operator.
+    Accepts kwargs for operator kwarg. Can be reused in a single DAG.
+
+    :param python_callable: Function to decorate
+    :type python_callable: Optional[Callable]
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. List/Tuples will unroll to xcom values
+        with index as key. Dict will unroll to xcom values with keys as XCom keys.
+        Defaults to False.
+    :type multiple_outputs: bool
+
+    """
+    def wrapper(f):
+        """
+        Python wrapper to generate PythonFunctionalOperator out of simple python functions.
+        Used for Airflow functional interface
+        """
+        _PythonFunctionalOperator.validate_python_callable(f)
+        kwargs['task_id'] = kwargs.get('task_id', None) or f.__name__
```

Comment:
Unimportant change but:

```suggestion
        kwargs.setdefault('task_id', f.__name__)
```